dnf/plugins/download/dnf-command-download.c

```diff
@@ -0,0 +1,324 @@
+/* dnf-command-download.c
+ *
+ * Copyright © 2020-2021 Daniel Hams <daniel.hams@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "dnf-command-download.h"
+#include "dnf-utils.h"
+
+/* For MAXPATHLEN */
+#include <sys/param.h>
+
+struct _DnfCommandDownload
+{
+  PeasExtensionBase parent_instance;
+};
+
+static void dnf_command_download_iface_init (DnfCommandInterface *iface);
+
+G_DEFINE_DYNAMIC_TYPE_EXTENDED (DnfCommandDownload,
+                                dnf_command_download,
+                                PEAS_TYPE_EXTENSION_BASE,
+                                0,
+                                G_IMPLEMENT_INTERFACE (DNF_TYPE_COMMAND,
+                                                       dnf_command_download_iface_init))
+
+static void
+dnf_command_download_init (DnfCommandDownload *self)
+{
+}
+
+inline static gboolean str_ends_with (const gchar * buf, const gchar * val)
+{
+  return strstr (buf,val) == (buf + (strlen (buf) - strlen (val)));
+}
+
+static gboolean dnf_command_download_rewriterepotosrc (gchar * repo_buf)
+{
+  if (str_ends_with (repo_buf, "-rpms")) {
+    strcat (repo_buf, "-source-rpms");
+    return TRUE;
+  }
+  else if (!str_ends_with (repo_buf, "-source")) {
+    strcat (repo_buf, "-source");
+    return TRUE;
+  }
+  return FALSE;
+}
+
+static gboolean
+dnf_command_download_enablesourcerepos (DnfContext *ctx,
+                                        DnfRepoLoader *repo_loader,
+                                        GError **error)
+{
+  g_autoptr(GError) local_error = NULL;
+  // A place where we can "extend" the repoid with an extension
+  gchar tmp_name[MAXPATHLEN];
+  g_autoptr(GPtrArray) all_repos = dnf_repo_loader_get_repos (repo_loader,error);
+
+  // We have to activate repos in a second loop to avoid breaking the iterator
+  // on the above repo list. We are "strdup"ing strings into it, so need
+  // free called on them.
+  g_autoptr(GPtrArray) repos_to_enable = g_ptr_array_new_full (all_repos->len,
+                                                               free);
+  // Look for each enabled repo if there is a corresponding source repo
+  // that we need to enable.
+  for (guint i = 0 ; i < all_repos->len; ++i ) {
```

Comment:
Thank you for reformating the code style.
Please, can you do another change.  `{`, `}` with `for`, `if`:  
```
for (;;)
  {
    func();
  }
```
To be honest. It is not my favorite style. But I want style consistency in the project.